.github/workflows/cluster_endtoend_vreplication_basic_self_hosted.yml

```diff
@@ -0,0 +1,37 @@
+# DO NOT MODIFY: THIS FILE IS GENERATED USING "make generate_ci_workflows"
+
+name: Cluster Self Hosted (vreplication_basic)
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: self-hosted
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Build Docker Image
+        run: docker build -f ./.github/dockerFiles/cluster_test_vreplication_basic/Dockerfile -t cluster_test_vreplication_basic:$GITHUB_SHA  .
+
+      - name: Run test
+        timeout-minutes: 30
+        run: docker run --name "cluster_test_vreplication_basic_$GITHUB_SHA" cluster_test_vreplication_basic:$GITHUB_SHA /bin/bash -c 'source build.env && go run test.go -keep-data=true -docker=false -print-log -follow -shard vreplication_basic -- -- --keep-data=true'
+
+      - name: Print Volume Used
+        if: ${{ always() }}
```

Comment:
nit: Can we omit the `always` condition?

Reply:
No we cannot omit the `always` condition. We want to print out the volume the docker container used irrespective of any failure in the previous step. If we remove the `always` condition then this would not run in case of testing failure and that would defeat the purpose of this step. Since it will be used only when someone tries to examine the logs which will largely happen in case of failures. Similarly we want to cleanup the docker container irrespective of failure

Reply:
Got it!